ios-trivia-game/ViewControllers/GameOptionsViewController.swift

```diff
@@ -10,17 +10,62 @@ import UIKit
 
 class GameOptionsViewController: UIViewController {
 
+    @IBOutlet weak var categoryPicker: UIPickerView!
+    @IBOutlet weak var numOfQuestionsPicker: UIPickerView!
+    
+    var categoryPickerData: [String] = [String]()
+    var numOfQuestionsPickerData: [String] = [String]()
+    
+    var numOfQuestions: Int?
+    
     override func viewDidLoad() {
         super.viewDidLoad()
 
         // Do any additional setup after loading the view.
+        getCategories()
+    }
+    
+    func getCategories() {
+        JServiceClient.instance.categories(count: 10, offset: 10, success: {(result: Any?) in
+            //print("my result : \(result)")
+            let triviaCategories = self.convertWithArray(dictionaries: result as! [NSDictionary])
+            for category in triviaCategories {
+                self.categoryPickerData.append(category.title!)
+            }
+            self.setupPickerData()
+            }, failure: { (error: Error?) in
+                    print("Error: \(error?.localizedDescription)")
+            })
+    }
+    
+    func convertWithArray(dictionaries: [NSDictionary]) -> [TriviaCategory] {
+        var categories = [TriviaCategory]()
+        
+        for dictionary in dictionaries {
+            let category = TriviaCategory(dictionary: dictionary)
+            categories.append(category)
+        }
+        
+        return categories
+    }
+    
+    func setupPickerData() {
```

Comment:
Kind of. I wanted to populate picker data after getting categories from server request.